packages/agoric-cli/lib/start.js

```diff
@@ -111,6 +204,283 @@ export default async function startMain(progname, rawArgs, powers, opts) {
     return ps;
   }
 
+  async function startLocalChain(profileName, startArgs, popts) {
+    const IMAGE = `agoric/agoric-sdk`;
+
+    const portNum = startArgs[0] === undefined ? CHAIN_PORT : startArgs[0];
+    if (`${portNum}` !== `${Number(portNum)}`) {
+      log.error(`Argument to local-chain must be a port number`);
+      return 1;
+    }
+    agServer += `-${portNum}`;
```

Comment:
Mutating a variable that's shared outside this function is weird. I see that it's declared early, and populated late (based on argv, and after all the subcommand definitions). And I know `startLocalChain` is only expected to be called once (indeed this mutation means it would break if called multiple times, creating a directory named `dev-26657-26658-26659` or something). But maybe use `const localAgServer = '${agServer}-${portNum}'` or similar, and consider refactoring the `agSolo`/`agServer` globals to put their definitions before their uses (maybe move the argv processing up to where `agSolo` is declared, and just leave the final `startFn(..)` call at the end?).

Reply:
I will redo that to be predictable.

Reply:
It's better, but still not perfect.  It's an improvement, so I'm not going to spend more time on this now.